### PR TITLE
Use SIL Open Font in example

### LINKED ISSUE #21553: Font-variant-alternates page suggests to pirate a copyrighted font
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-alternates

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-alternates#examples

### What information was incorrect, unhelpful, or incomplete?

> Note: You need to install the OpenType font Leitura Display Swashes for this example to work. You can find a few free versions for testing purposes, for example from [fontsgeek.com](https://fontsgeek.com/fonts/Leitura-Display-Swashes).

### What did you expect to see?

I suggest to use an open-source font for swashes and stylistic set examples, instead of publishing a suggestion to break licensing agreements.

### Do you have any supporting links, references, or citations?

Leitura is a font that can be licensed from DSType. While the internet may have copies of it, it doesn't mean that there's a testing license granted. https://www.dstype.com/fonts/leitura-display-two/glyphs/swashes/all-glyphs

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/font-variant-alternates`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-alternates
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/font-variant-alternates/index.md
* Last commit: https://github.com/mdn/content/commit/fb2af1f15456199685a9f4fbaf8c9d003a2bf91f
* Document last modified: 2022-09-27T02:27:51.000Z

</details>

### PR DESCRIPTION
**Pull request by Elchi3**
### Description

This PR fixes https://github.com/mdn/content/issues/21553. I don't much about fonts. I picked a font from this list https://github.com/mdn/content/issues/21553#issuecomment-1279218134 and removed the reference to fontsgeek.com.

### Motivation

Use of open source fonts instead of a suggestion to break licensing agreements.

### Additional details

None

### Related issues and pull requests

None

### CODE REVIEW

File: files/en-us/web/css/font-variant-alternates/index.md [hunk 1]
```diff
@@ -102,14 +102,14 @@ p {
...
-  font-family: Leitura Display Swashes;
+  font-family: Bellota Text;
   font-variant-alternates: swash(fancy);
 }
 ```
 
 #### Result
 
-> **Note:** You need to install the OpenType font _Leitura Display Swashes_ for this example to work. You can find a few free versions for testing purposes, for example from [fontsgeek.com](https://fontsgeek.com/fonts/Leitura-Display-Swashes).
+> **Note:** You need to install the OpenType font _Bellota Text_ for this example to work. The font is licensed under the [SIL Open Font License Version 1.1](http://scripts.sil.org/OFL) and can be found at <https://github.com/kemie/Bellota-Font>. More fonts can be found in the [Google Font corpus](https://github.com/google/fonts).
```
Comment (drott) [inline files/en-us/web/css/font-variant-alternates/index.md:112]: While at it, could the test be changed to not require an install of the font (this sounds a bit dated given _web fonts,_ but instead embedded it as a web font either by referring to a Google Fonts URL, or a local copy of Bellota in a `src: url(...Bellota filename...)`?

**Review by drott — COMMENTED** [review]
Thank you @Elchi3 for taking action - one suggestion below.

File: files/en-us/web/css/font-variant-alternates/index.md (hunk 1 above)
Comment (Elchi3) [reply files/en-us/web/css/font-variant-alternates/index.md:112]: I believe using Google Fonts or local fonts doesn't work right now in MDN examples. See https://github.com/mdn/yari/issues/5727#issuecomment-1236449710

I will ask around to see if anyone knows more.

Comment (schalkneethling) [reply files/en-us/web/css/font-variant-alternates/index.md:112]: Is the inability to use Google fonts because of the CSP?

Comment (wbamberg) [reply files/en-us/web/css/font-variant-alternates/index.md:112]: Google Fonts works: https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode#using_mix-blend-mode_with_text . It's local fonts that don't.

Comment (hamishwillee) [reply files/en-us/web/css/font-variant-alternates/index.md:112]: Google fonts work well, but if you have a problem it is not always easy to tell whether the font file you generated from their API is the problem or your code. Hence the yari issue - in that case I had a known good font I could have used and that would have saved me a lot of time.

Comment (Elchi3) [reply files/en-us/web/css/font-variant-alternates/index.md]: Thanks, loading a Google font works. I updated this.

**Review by drott — COMMENTED** [review]
Non-owner LGTM, thanks for updating this page.

**Comment by wbamberg** [comment]
It's great to see a live sample for this, but I'm not sure how this shows the use of `font-variant-alternates` specifically.

The first example shows the effect of not setting `font-family` at all, while the second shows the effect of using `font-family: "Bellota";` _and_ `font-variant-alternates`. So it's not clear whether the difference is due to the different font family or due to the `font-variant-alternates` declaration.

It looks like you can get just the same effect without `font-variant-alternates`:

Screen Shot 2022-10-18 at 9 01 24 AM


...so I'm not sure what it is adding here. Am I missing something?

**Comment by Elchi3** [comment]
I'm seeing the same thing as Will. It is strange that the example has two differences, the` font-family` and `font-variant-alternates`. I guess it should be changed so that `font-variant-alternates` is the only difference. 

It seems like this is only implemented in Firefox and I'm not sure if I see a visual difference in the rendering either. I'm using macOS. Is it a compat thing? Do we need to choose a different font? Is the code wrong?

**Comment by wbamberg** [comment]
> I'm seeing the same thing as Will. It is strange that the example has two differences, the` font-family` and `font-variant-alternates`. I guess it should be changed so that `font-variant-alternates` is the only difference.
> 
> It seems like this is only implemented in Firefox and I'm not sure if I see a visual difference in the rendering either. I'm using macOS. Is it a compat thing? Do we need to choose a different font? Is the code wrong?

Since neither of us can apparently figure out how to use this, I think it's fair to say that the documentation is not very helpful :). Maybe we should merge this PR as is (since it does resolve the original issue about recommending piracy, which is not good) and have a separate issue about this page? Note that this is [part of Interop 2022 and is marked as "done"](https://github.com/openwebdocs/project/issues/137), presumably because the page exists - maybe we should also review pages that exist and decide whether we can improve them.

**Comment by drott** [comment]
`font-variant-alternates` is currently only supported in [Safari and FF](https://caniuse.com/?search=font-variant-alternates). This may help with testing. Also, if a font does not show differing results, you may want to check the font features in https://wakamaifondue.com/beta/ and if there's not the desired feature level for alternates, try a different one from the list above.

**Comment by Elchi3** [comment]
I've tried like 15 fonts now and tested them in Firefox and Safari and I don't see any differences. :(

**Comment by drott** [comment]
Please check the font in Wakamai Fondue. Swash glyph alternates only exists for certain glyphs, usually S or A or something that has a curly part at the end or beginning of the letter. Wakamai Fondue will show you for which glyphs it has such swashes, then you can adapt the test text accordingly.

**Comment by Elchi3** [comment]
I've tried more and still don't see anything. Do I need more code like `font-feature-settings: "swsh" 1;`?

Also, I don't know where on Wakamai Fondue it tells me that a font does swashes. I found the "Sansita Swashed" family and it does look like some letters are swashed but the CSS snippet doesn't change anything there either.

(Sorry to be so slow here, this isn't my area of expertise)

**Comment by drott** [comment]
If you use https://fonts.google.com/specimen/MonteCarlo it definitely has swash glyphs for the following glyphs (see below).

For example, for a capital A, without `swsh`, and with `swsh` they look as follows (slightly truncated, I was testing with `hb-view`).
`$ hb-view MonteCarlo-Regular.ttf --features "swsh=1" "A"` vs
`hb-view MonteCarlo-Regular.ttf  "A"`

A_noswash
A_swash

So with this font, and using the glyphs from the list below, you'll be able to see difference.



```
        <SingleSubst index="0">
          <Substitution in="A" out="A.3"/>
          <Substitution in="B" out="B.3"/>
          <Substitution in="C" out="C.3"/>
          <Substitution in="D" out="D.3"/>
          <Substitution in="E" out="E.3"/>
          <Substitution in="F" out="F.3"/>
          <Substitution in="G" out="G.3"/>
          <Substitution in="H" out="H.3"/>
          <Substitution in="I" out="I.3"/>
          <Substitution in="J" out="J.3"/>
          <Substitution in="K" out="K.3"/>
          <Substitution in="L" out="L.3"/>
          <Substitution in="M" out="M.3"/>
          <Substitution in="N" out="N.3"/>
          <Substitution in="O" out="O.3"/>
          <Substitution in="P" out="P.3"/>
          <Substitution in="Q" out="Q.3"/>
          <Substitution in="R" out="R.3"/>
          <Substitution in="S" out="S.3"/>
          <Substitution in="T" out="T.3"/>
          <Substitution in="U" out="U.3"/>
          <Substitution in="V" out="V.3"/>
          <Substitution in="W" out="W.3"/>
          <Substitution in="X" out="X.3"/>
          <Substitution in="Y" out="Y.3"/>
          <Substitution in="Z" out="Z.3"/>
          <Substitution in="b" out="b.5"/>
          <Substitution in="d" out="d.16"/>
          <Substitution in="eight" out="crown1"/>
          <Substitution in="f" out="f.3"/>
          <Substitution in="five" out="embellish6"/>
          <Substitution in="four" out="embellish5"/>
          <Substitution in="g" out="g.7"/>
          <Substitution in="h" out="h.8"/>
          <Substitution in="k" out="k.7"/>
          <Substitution in="l" out="l.7"/>
          <Substitution in="nine" out="crown2"/>
          <Substitution in="one" out="embellish2"/>
          <Substitution in="p" out="p.4"/>
          <Substitution in="q" out="q.3"/>
          <Substitution in="s" out="s.5"/>
          <Substitution in="seven" out="embellish8"/>
          <Substitution in="six" out="embellish7"/>
          <Substitution in="t" out="t.8"/>
          <Substitution in="t_t" out="t_t.1"/>
          <Substitution in="three" out="embellish4"/>
          <Substitution in="two" out="embellish3"/>
          <Substitution in="y" out="y.6"/>
          <Substitution in="zero" out="embellish1"/>
        </SingleSubst>
```

**Comment by drott** [comment]
Yes, same here, but your test page works with the regular MonteCarlo from the download, Google Fonts seems to strip the swash feature, compare: https://roettsch.es/swash.html where I am embedding the full font with your test page.

**Comment by Elchi3** [comment]
Ahhh finally seeing the swashes as well! :) 
hm, now the problem is that MDN doesn't support local fonts right now, as discussed in the conversation above. 

Maybe it is also worth filing a bug with Google Fonts.

**Comment by wbamberg** [comment]
Thanks for your help understanding this, @drott .

> Google Fonts seems to strip the swash feature

That was the critical fact we were missing :). I spent a bunch of time trying to get swashes to work with MonteCarlo last week, but didn't try locally.

For this PR, perhaps we could:
- stop making this a live sample but instead make it use a local font and tell people where they can get a version of the font that will work
- note that Google Fonts strips swashes

More work, but we could also, in a follow-up:

- find a `font-variant-alternates` feature that _is_ supported in a Google Fonts-hosts font, and include a live sample of that. It would be good anyway to have more than one feature example here.
- we should also include something that fleshes out "The parameter is a font-specific name mapped to a number." in a practically helpful way, potentially with examples of what you could specify with a couple of fonts, or how you could figure out what to specify, given a font you want to use.

**Comment by drott** [comment]
Can the full font asset be fetched or placed somewhere where the inline
example can fetch it? how do you do examples with images usually?

**Comment by wbamberg** [comment]
> Can the full font asset be fetched or placed somewhere where the inline example can fetch it? how do you do examples with images usually?

Unfortunately not, because of https://github.com/mdn/yari/issues/5727.

**Comment by drott** [comment]
> > Can the full font asset be fetched or placed somewhere where the inline example can fetch it? how do you do examples with images usually?
> 
> Unfortunately not, because of [mdn/yari#5727](https://github.com/mdn/yari/issues/5727).

That issue mentions using "our own CDN" as an alternative - so maybe manually placing it there, then referring to a CDN url for the asset might be an option? I am not sure how complicated that is.

**Comment by wbamberg** [comment]
> > > Can the full font asset be fetched or placed somewhere where the inline example can fetch it? how do you do examples with images usually?
> > 
> > 
> > Unfortunately not, because of [mdn/yari#5727](https://github.com/mdn/yari/issues/5727).
> 
> That issue mentions using "our own CDN" as an alternative - so maybe manually placing it there, then referring to a CDN url for the asset might be an option? I am not sure how complicated that is.

I believe that's not a reference to an immediately usable alternative, but a suggestion for a possible way to solve it (i.e. it's also a project). I think that if we want to put time into building a solution, then the local files approach would be better in various ways.

**Comment by Elchi3** [comment]
In order to move forward here, I documented how to best get to swashes using a self loaded font.

I removed the example rendering because MDN is unable to render it until something in https://github.com/mdn/yari/issues/5727 happened. :(

**Review by wbamberg — APPROVED** [review]
👍 thank you Florian. I think this is the best we can do for now :(

**Comment by wbamberg** [comment]
Thanks to https://github.com/mdn/yari/pull/7605, we now have a proper fix for this: https://github.com/mdn/content/pull/27206.

**Pull request merged**